### PR TITLE
chore(deps): combine Dependabot updates

### PR DESCRIPTION
**Pull request by NormannK**
Combines the currently open Dependabot dependency updates into one PR.

Included updates:
- cachebox 6.2.2 → 6.2.5 (#1248)
- tzfpy 1.3.2 → 1.3.3 (#1245)
- pydantic-settings 2.14.2 → 2.15.0 (#1246)
- mypy 2.3.0 → 2.3.1 (#1247)
- commitizen 4.16.5 → 4.17.1 (#1249)
- GitPython 3.1.58 → 3.1.59 (#1251)
- types-PyYaml 6.0.12.20260724 → 6.0.12.20260815 (#1252)
- pre-commit 4.6.1 → 4.6.2 (#1253)

Follow-up fixes from review:
- regenerated `uv.lock` so locked/frozen installs match the updated dependency pins
- synchronized `.pre-commit-config.yaml` with mypy 2.3.1 and types-PyYaml 6.0.12.20260815

The PR now changes only `pyproject.toml`, `.pre-commit-config.yaml`, and `uv.lock`, and is based on current `main`.

### CODE REVIEW

**Comment by NormannK** [comment]
### Significant changes / EOS impact

The dependency bumps are mostly patch/tooling updates, but a few are worth explicit review:

- **`pydantic-settings` 2.14.2 → 2.15.0 — highest runtime impact.** EOS directly subclasses `pydantic_settings.BaseSettings` for `SettingsEOS` / `ConfigEOS`, with `case_sensitive` left at its default. In 2.15.0, `case_sensitive` now also applies to init kwargs and config-file sources, so top-level setting keys are now matched case-insensitively by default where those sources previously did not behave that way. This can change how differently-cased config keys are accepted/resolved. The release also adds warnings for unresolved forward references and changes strict non-JSON env-value failures to `ValidationError`. **Recommended:** exercise JSON config loading, init/update paths, env overrides, and config round-trips.

- **`tzfpy` 1.3.2 → 1.3.3 — runtime correctness change.** EOS uses `tzfpy.get_tz()` in `to_timezone()` and exposes the result through `GeneralSettings.timezone`. Queries exactly on timezone polygon borders now resolve instead of returning an empty result, and the `America/Argentina/Ushuaia` boundary is corrected. This can intentionally change timezone output for users on/near affected boundaries.

- **`cachebox` 6.2.2 → 6.2.5 — runtime cache correctness/safety.** EOS uses `cachebox.LRUCache` and `cachebox.cached` for the energy-management cache. The update fixes iterator lifetime safety, LRU iterator invalidation when reads promote entries, and a potential `setdefault_with` locking issue. EOS does not appear to rely on those edge cases directly, so this is expected to be low-risk and mostly corrective; existing cache tests are the relevant regression coverage.

- **`GitPython` 3.1.58 → 3.1.59 — dev/tooling security hardening.** This release blocks file-reading Git options, separate git-directory use during clone, and hardens config parsing. It appears to be a dev/docs dependency rather than EOS runtime code, but CI/tooling that intentionally passes unusual Git options could be affected.

- **`pre-commit` 4.6.1 → 4.6.2 — dev-only bug fix.** Fixes a regression in Node-language hooks using npm 11.x build scripts.

- **`mypy` 2.3.1, `commitizen` 4.17.1, and `types-PyYaml` stub update** are tooling/type-checking changes with no expected EOS runtime behavior change.

Overall, the main compatibility focus should be **configuration handling (`pydantic-settings`)**, followed by **timezone edge cases (`tzfpy`)**. The remaining updates are primarily correctness, security, or developer-tooling fixes.

File: pyproject.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ classifiers = [
 dependencies = [
     "babel==2.18.0",
     "beautifulsoup4==4.15.0",
-    "cachebox==6.2.2",
+    "cachebox==6.2.5",
```
[the remark this answers is not in the document]
Comment (NormannK) [reply pyproject.toml:19]: Fixed. `uv.lock` was regenerated from the updated `pyproject.toml` with `uv lock` on the PR branch, so both the project metadata and resolved package entries now match all eight dependency bumps. The generated lockfile is committed in `chore(deps): refresh uv lock`.

File: pyproject.toml [hunk 2]
```diff
@@ -59,21 +59,21 @@ dev = [
     # - commitizen - sync with requirements-dev.txt (if on pypi)
     #
     # !!! Sync .pre-commit-config.yaml with these dependencies !!!
-    "pre-commit==4.6.1",
-    "mypy==2.3.0",
+    "pre-commit==4.6.2",
+    "mypy==2.3.1",
```
[the remark this answers is not in the document]
Comment (NormannK) [reply pyproject.toml:63]: Fixed. `.pre-commit-config.yaml` is now synchronized with the dev dependency pins: `mirrors-mypy` uses `v2.3.1` and the hook’s `types-PyYaml` additional dependency is `6.0.12.20260815`. This keeps the pre-commit mypy job aligned with `pyproject.toml`.
